packages/design-system-react/src/components/button/Button.types.ts

```diff
@@ -0,0 +1,36 @@
+import type { ButtonPrimaryProps } from '../button-primary';
+import type { ButtonSecondaryProps } from '../button-secondary';
+import type { ButtonTertiaryProps } from '../button-tertiary';
+
+export enum ButtonVariant {
+  /**
+   * Primary button variant - used for primary actions
+   */
+  Primary = 'primary',
+  /**
+   * Secondary button variant - used for secondary actions
+   */
+  Secondary = 'secondary',
+  /**
+   * Tertiary button variant - used for tertiary-like actions
+   */
+  Tertiary = 'tertiary',
+}
+
+export type ButtonProps = {
+  /**
+   * Optional prop to control the variant of the Button
+   * @default ButtonVariant.Primary
+   */
+  variant?: ButtonVariant;
+} & (
+  | (Omit<ButtonPrimaryProps, 'ref'> & {
+      variant?: ButtonVariant.Primary;
+    })
+  | (Omit<ButtonSecondaryProps, 'ref'> & {
+      variant?: ButtonVariant.Secondary;
+    })
+  | (Omit<ButtonTertiaryProps, 'ref'> & {
+      variant?: ButtonVariant.Tertiary;
+    })
+);
```

Comment:
so you cant use ref with Button?

Reply:
Good question! We can definitely use refs with Button! We're using React's `forwardRef` pattern which explicitly enables ref support. The `Omit<..., 'ref'>` in the types is specifically there to avoid type conflicts since `forwardRef` handles the ref prop separately from other props.

You can see this working in our test suite under the "Ref Forwarding" describe block:
https://github.com/MetaMask/metamask-design-system/blob/db1c861ecc071bf554af791c0aceab40bbbbe56c/packages/design-system-react/src/components/button/Button.test.tsx#L238-L263